motionplan/utils.go

```diff
@@ -98,3 +45,247 @@ func (r *resultPromise) result() ([]node, error) {
 	}
 	return planReturn.steps, nil
 }
+
+// linearizedFrameSystem wraps a framesystem, allowing conversion in a known order between a map[string][]inputs and a flat array of floats,
+// useful for being able to call IK solvers against framesystems.
+type linearizedFrameSystem struct {
```

Comment:
Is this the solverframe code?

Reply:
No. This simply allows things to call nlopt, which requires a vector of floats.

What little code survives from `solverFrame`, is now `motionChain`

Reply:
These new concepts are 95% of this file, if we remove them then there will only be like 30 lines left. They seem appropriate for misc stuff, in that this file is filled with assorted private helper structs?